---
title: "BAYESPRISM + SPACEFOLD"
output: github_document
editor_options: 
  markdown: 
    wrap: 72
---

## BAYESPRISM + SPACEFOLD -

De-convolution and pseudo-space projection.

References and links-

### **BayesPrism**

-   Original paper -(DOI- <https://doi.org/10.1038/s43018-022-00356-3>),

-   Github page and documentation
    (<https://github.com/Danko-Lab/BayesPrism>)

### **SpaceFold**

-   Original paper - (DOI-
    <https://doi.org/10.1016/j.stem.2022.05.007>),

-   Github page and documentation
    (<https://github.com/dpeerlab/SpaceFold_paper>)

| FIGURE NO / LINK                 | DESCRIPTION                                                               |
|----------------------|--------------------------------------------------|
| [FIGURE 5D - HEALTHY SKIN]       | Space-fold projection - healthy                                           |
| [FIGURE 5D - NON -LESIONAL SKIN] | Space-fold projection - non-lesional                                      |
| [FIGURE 5D - LESIONAL SKIN]      | Space-fold projection - lesional                                          |
| [FIGURE 7A]                      | Space-fold projection - Mild disease (non-lesional & lesional)            |
| [FIGURE 7B]                      | Space-fold projection - Moderate-severe disease (non-lesional & lesional) |

```{r}
library(Seurat)
library(tidyverse)
library("devtools")
library(TED)
library(ggsci)
```

```{r}
skin_data.hm.sct <- readRDS("/Volumes/Extreme Pro/GITHUB-DATA/ST-DATA/PSORIASIS-DATA/RDS-Files/ALL_SPATIAL_SAMPLES(HM_BATCH_CORRECTED).RDS")
```

*Import processed (Seurat object) single cell RNA (scRNA) data from
Hughes et al (Dataset 1)*

Psoriatic skin data

```{r,eval=FALSE}
travis.psoriatic_data <- readRDS("/Volumes/Extreme Pro/GITHUB-DATA/SC-RNA-DATA/TRAVIS-DATA/RDS-Files/TRAVIS_PS_DATA.RDS")
```

Healthy skin data

```{r,eval=FALSE}
travis_normal_skin <- readRDS("/Volumes/Extreme Pro/GITHUB-DATA/SC-RNA-DATA/TRAVIS-DATA/RDS-Files/travis_NM_skin_scRNA.RDS")
```

Process single cell data

```{r,eval=FALSE}
source("/Volumes/Extreme Pro/GITHUB-DATA/SCRIPTS/spacefold/process_nm_input.R")
source("/Volumes/Extreme Pro/GITHUB-DATA/SCRIPTS/spacefold/process_ps_input.R")
```

```{r,eval=FALSE}
load("ref.sig.ps.rdata")
```

```{r}
##write a function for convenience
### Please adjust parameters as needed
run_bayes_prism <- function(st.data,sc.data){
  ref.sc.data <- t(as.matrix(sc.data@assays$RNA@counts))
  
  ref.data.filtered <- cleanup.genes(ref.dat=ref.sc.data,species="hs",gene.type=c("RB","chrX","chrY","chrM"),input.type="scRNA",exp.cells = 10)
  
  specific.cell.type.labels <- sc.data@meta.data$Specific_CellType
  cell.type.labels <- sc.data@meta.data$CellType
  
  dat.st <- t(as.matrix(GetAssayData(object = st.data, slot = "counts",assay ="Spatial")) %>% data.frame() %>% filter(rowSums(across(where(is.numeric)))!=0) )
  
  ted.output <- run.Ted(ref.dat = ref.data.sig.filtered,
 							X= dat.st,
 							cell.type.labels = cell.type.labels.filtered,
 							cell.subtype.labels = specific.cell.type.labels.filtered,
 							input.type = "scRNA",
 							n.cores =31,
 							seed=123,
 							outlier.fraction=0.9)
  return(ted.output)
}
```

Split the Seurat ST data into three main groups-

1.  Lesional

2.  Non-Lesional

3.  Healthy / Normal skin

Following that, we run BayesPrism pipeline on all three groups,
independently.

[(NOTE: This process requires a long run time and high memory
usage)]{.underline}

```{r}
ps_lesional_skin.st <- subset(skin_data.hm.sct,DISEASE_STATUS %in% c("Lesional"))
ps_non_lesional_skin.st <- subset(skin_data.hm.sct,DISEASE_STATUS %in% c("Non-Lesional"))
healthy_skin.st <- subset(skin_data.hm.sct,DISEASE_STATUS %in% c("Healthy skin"))

```

```{r,eval=FALSE}

ps_lesional.st.counts <- as.matrix(GetAssayData(object = ps_lesional_skin.st, slot = "counts",assay ="Spatial")) %>% data.frame() %>% filter(rowSums(across(where(is.numeric)))!=0) 

write_tsv(ps_lesional.st.counts,file="ps_lesional.st.counts.tsv")

ps.lesional.ted <- run_bayes_prism(ps_lesional.st.counts,travis.psoriatic_data)
```

```{r,eval=FALSE}

ps_non_lesional_skin.st.counts <- as.matrix(GetAssayData(object = ps_non_lesional_skin.st, slot = "counts",assay ="Spatial")) %>% data.frame() %>% filter(rowSums(across(where(is.numeric)))!=0) 

write.table(ps_non_lesional_skin.st.counts,file="ps_non_lesional_skin.st.counts.tsv",row.names = TRUE,)

ps.non_lesional.ted <- run_bayes_prism(ps_non_lesional_skin.st,travis.psoriatic_data)
```

```{r,eval=FALSE}

healthy_skin.st.counts <- as.matrix(GetAssayData(object = healthy_skin.st, slot = "counts",assay ="Spatial")) %>% data.frame() %>% filter(rowSums(across(where(is.numeric)))!=0)

write_tsv(healthy_skin.st.counts,file="healthy_skin.st.counts.tsv")

healthy_skin.st.ted <- run_bayes_prism(healthy_skin.st.counts,travis_normal_skin)
```

## PS SAMPLES - grouped by severity ratings

Split the Seurat ST data into different groups on the basis of disease
severity i.e. PASI score and disease status . Mild group samples have a
PASI score\<12 and Moderate-severe group samples have a PASI score \>=
12.

The following four groups were created:-

1.  Mild Lesional

2.  Mild non-Lesional

3.  Moderate-severe Lesional

4.  Moderate-severe non-Lesional

## Subset the seurat data into the 4 groups

```{r}
ps_lesional_skin.st_mild <- subset(skin_data.hm.sct,DISEASE_STATUS %in% c("Lesional")) %>% subset(PASI<12)
ps_lesional_skin.st_severe <- subset(skin_data.hm.sct,DISEASE_STATUS %in% c("Lesional")) %>% subset(PASI>=12)
ps_non_lesional_skin.st_mild <- subset(skin_data.hm.sct,DISEASE_STATUS %in% c("Non-Lesional")) %>% subset(PASI<12)
ps_non_lesional.st_severe <- subset(skin_data.hm.sct,DISEASE_STATUS %in% c("Non-Lesional")) %>% subset(PASI>=12)
```

##Run BayesPrism on each group -

### MILD LESIONAL SAMPLES

```{r,eval=FALSE}
ps_lesional_skin.st_mild.counts <- as.matrix(GetAssayData(object = ps_lesional_skin.st_mild, slot = "counts",assay ="Spatial")) %>% data.frame() %>% filter(rowSums(across(where(is.numeric)))!=0)

write_tsv(ps_lesional_skin.st_mild.counts,file="ps_lesional_skin.st_mild.counts.tsv")

ps_lesional_skin.st_mild.ted <- run_bayes_prism(ps_lesional_skin.st_mild.counts,travis.psoriatic_data)
```

### MODERATE-SEVERE LESIONAL SAMPLES

```{r,eval=FALSE}
ps_lesional_skin.st_severe.counts <- as.matrix(GetAssayData(object = ps_lesional_skin.st_severe, slot = "counts",assay ="Spatial")) %>% data.frame() %>% filter(rowSums(across(where(is.numeric)))!=0)

write_tsv(ps_lesional_skin.st_severe.counts,file="ps_lesional_skin.st_severe.counts.tsv")

ps_lesional_skin.st_severe.ted <- run_bayes_prism(ps_lesional_skin.st_severe.counts,travis.psoriatic_data)
```

### MILD NON-LESIONAL SAMPLES

```{r,eval=FALSE}

ps_non_lesional_skin.st_mild.counts <- as.matrix(GetAssayData(object = ps_non_lesional_skin.st_mild, slot = "counts",assay ="Spatial")) %>% data.frame() %>% filter(rowSums(across(where(is.numeric)))!=0)

write_tsv(ps_non_lesional_skin.st_mild.counts,file="ps_non_lesional_skin.st_mild.counts.tsv")

ps_non_lesional_skin.st_mild.ted <- run_bayes_prism(ps_non_lesional_skin.st_mild.counts,travis.psoriatic_data)
```

### MODERATE-SEVERE NON-LESIONAL SAMPLES

```{r,eval=FALSE}

ps_non_lesional.st_severe.counts <- as.matrix(GetAssayData(object = ps_non_lesional.st_severe, slot = "counts",assay ="Spatial")) %>% data.frame() %>% filter(rowSums(across(where(is.numeric)))!=0)

write_tsv(ps_non_lesional.st_severe.counts,file="ps_non_lesional.st_severe.counts.tsv")

ps_non_lesional.st_severe.ted <- run_bayes_prism(ps_non_lesional.st_severe.counts,travis.psoriatic_data)
```

### LOAD PROCESSED DATA FOR SPACE-FOLD (Pre-processed by running BayesPrism pipeline as described above)

```{r}
ps.lesional.ted <- readRDS("/Volumes/Extreme Pro/GITHUB-DATA/BAYESPRISM-DATA/BAYES_PRISM_OUTPUT/Archive/ps.lesional.ted.RDS")
ps.non_lesional.ted <- readRDS("/Volumes/Extreme Pro/GITHUB-DATA/BAYESPRISM-DATA/BAYES_PRISM_OUTPUT/Archive/ps.non_lesional.ted.RDS")
healthy_skin.st.ted <- readRDS("/Volumes/Extreme Pro/GITHUB-DATA/BAYESPRISM-DATA/BAYES_PRISM_OUTPUT/Archive/healthy_skin.st.ted.RDS")
```

```{r}
#load source files
source.dir <- "/Volumes/Extreme Pro/GITHUB-DATA/SCRIPTS/spacefold/"
source (paste(source.dir,"bp.merge.function.R",sep=""))
source (paste(source.dir,"bp.utility.functions.R",sep=""))
source (paste(source.dir,"project.exp.functions.R",sep=""))
source (paste(source.dir,"run.phate.functions.R",sep=""))
#source("../../SPACE_FOLD_CODE/run.phate.functions.R")
```

```{r}
structural.cells<- c("KC","HairFollicle","VSMC","Lymphatic","Endo","Fib")
immune.cells <- c("Langerhans","Myeloid","T","B")

healthy.order <- c("KC","HairFollicle","Melanocyte","VSMC","Lymphatic","Endo","Fibro","Langerhans","Myeloid","Mast","T")
ps.order <- c("KC","HairFollicle","Melanocyte","VSMC","Lymphatic","Endo","Fibro","Langerhans","Myeloid","Mast","T","B")
```

```{r}
healthy_skin.st.ted$para$feature <- healthy_skin.st.ted$para$X
```

```{r}
meta <- as.data.frame(healthy_skin.st@meta.data) %>% rownames_to_column("barcode")
rownames(healthy_skin.st.ted$para$X) <- meta$barcode
healthy_skin.st.ted <- add.meta(bp.obj= healthy_skin.st.ted, meta =meta)
```

### FIGURE 5D - HEALTHY SKIN

```{r}
#compute Znk (fraction of reads in each cell type)
healthy_skin.st.ted <- compute.Znk(healthy_skin.st.ted)

#normalize by Znk
healthy_skin.st.ted <- norm.by.sf(healthy_skin.st.ted)

#determine background level, using final theta
healthy_skin.st.ted <- compute.background.level (healthy_skin.st.ted, which.theta='final')

#run phate
# specify anchorCellTypes to select cell types (if you need to held out a group of cell types) 
healthy_skin.st.ted <- phate.bp (healthy_skin.st.ted, if.invert=FALSE)

#plot cell type distribution along the spacefold axis
plot.beeswarm.no.pdf(healthy_skin.st.ted, pdf.prefix="HEALTHY_SKIN_SPACEFOLD_OUTPUT(0.90_cutoff)",q.cut = 0.90,cell.type.order = healthy.order)

#plot cell type distribution along the spacefold axis
#plot.beeswarm.no.pdf(healthy_skin.st.ted, pdf.prefix="HEALTHY_SKIN_SPACEFOLD_OUTPUT(0.95_cutoff)",q.cut = 0.95,cell.type.order = healthy.order)
```

MATCH THE BARCODES

```{r}
ps.lesional.ted$para$feature <- ps.lesional.ted$para$X
```

```{r}
meta <- as.data.frame(ps_lesional_skin.st@meta.data) %>% rownames_to_column("barcode")
rownames(ps.lesional.ted$para$X) <- meta$barcode
ps.lesional.ted <- add.meta(bp.obj= ps.lesional.ted, meta =meta)
```

### FIGURE 5D - LESIONAL SKIN

```{r}
#compute Znk (fraction of reads in each cell type)
ps.lesional.ted <- compute.Znk(ps.lesional.ted)

#normalize by Znk
ps.lesional.ted <- norm.by.sf(ps.lesional.ted)

#determine background level, using final theta
ps.lesional.ted <- compute.background.level (ps.lesional.ted, which.theta='final')

#run phate
# specify anchorCellTypes to select cell types (if you need to held out a group of cell types) 
ps.lesional.ted <- phate.bp (ps.lesional.ted, if.invert=FALSE)

#plot cell type distribution along the spacefold axis
#plot.beeswarm.no.pdf(ps.lesional.ted, pdf.prefix="PS_LESIONAL_SKIN_SPACEFOLD_OUTPUT(0.95_cutoff)",q.cut = 0.90,cell.type.order = ps.order)

#plot cell type distribution along the spacefold axis
plot.beeswarm.no.pdf(ps.lesional.ted, pdf.prefix="PS_LESIONAL_SKIN_SPACEFOLD_OUTPUT(0.90_cutoff)",q.cut = 0.90,cell.type.order = ps.order)

```

```{r}
ps.non_lesional.ted$para$feature <- ps.non_lesional.ted$para$X
```

MATCH THE BARCODES

```{r}
meta <- as.data.frame(ps_non_lesional_skin.st@meta.data) %>% rownames_to_column("barcode")
rownames(ps.non_lesional.ted$para$X) <- meta$barcode
ps.non_lesional.ted <- add.meta(bp.obj= ps.non_lesional.ted, meta =meta)
```

### FIGURE 5D - NON -LESIONAL SKIN

```{r}
#compute Znk (fraction of reads in each cell type)
ps.non_lesional.ted <- compute.Znk(ps.non_lesional.ted)

#normalize by Znk
ps.non_lesional.ted <- norm.by.sf(ps.non_lesional.ted)

#determine background level, using final theta
ps.non_lesional.ted <- compute.background.level (ps.non_lesional.ted, which.theta='final')

#run phate
# specify anchorCellTypes to select cell types (if you need to held out a group of cell types) 
ps.non_lesional.ted <- phate.bp (ps.non_lesional.ted, if.invert=FALSE,renorm.to.one = TRUE)

#plot cell type distribution along the spacefold axis
#plot.beeswarm.no.pdf(ps.non_lesional.ted, pdf.prefix="PS_NON_LESIONAL_SKIN_SPACEFOLD_OUTPUT(0.95_cutoff)",q.cut = 0.95,cell.type.order = ps.order)

#plot cell type distribution along the spacefold axis
plot.beeswarm.no.pdf(ps.non_lesional.ted, pdf.prefix="PS_NON_LESIONAL_SKIN_SPACEFOLD_OUTPUT(0.90_cutoff)",q.cut = 0.90,cell.type.order = ps.order)
```

## LOAD SUB-LESIONAL SKIN DATA - BY SEVERITY

```{r}
ps_lesional_skin.st_mild.ted <- readRDS("/Volumes/Extreme Pro/GITHUB-DATA/BAYESPRISM-DATA/BAYES_PRISM_OUTPUT/Archive/ps_lesional_skin.st_mild.ted.RDS")
ps_lesional_skin.st_severe.ted <- readRDS("/Volumes/Extreme Pro/GITHUB-DATA/BAYESPRISM-DATA/BAYES_PRISM_OUTPUT/Archive/ps_lesional_skin.st_severe.ted.RDS")
```

```{r}
ps_lesional_skin.st_mild.ted$para$feature <- ps_lesional_skin.st_mild.ted$para$X
```

MATCH THE BARCODES

```{r}
meta <- as.data.frame(ps_lesional_skin.st_mild@meta.data) %>% rownames_to_column("barcode")
rownames(ps_lesional_skin.st_mild.ted$para$X) <- meta$barcode
ps_lesional_skin.st_mild.ted <- add.meta(bp.obj= ps_lesional_skin.st_mild.ted, meta =meta)
```

## **FIGURE 7A** 

#### **FIGURE 7A -** LESIONAL & MILD

#### SPACE-FOLD PROJECTION (LESIONAL SKIN - MILD GROUP)

```{r}
#compute Znk (fraction of reads in each cell type)
ps_lesional_skin.st_mild.ted <- compute.Znk(ps_lesional_skin.st_mild.ted)

#normalize by Znk
ps_lesional_skin.st_mild.ted <- norm.by.sf(ps_lesional_skin.st_mild.ted)

#determine background level, using final theta
ps_lesional_skin.st_mild.ted <- compute.background.level (ps_lesional_skin.st_mild.ted, which.theta='final')

#run phate
# specify anchorCellTypes to select cell types (if you need to held out a group of cell types) 
ps_lesional_skin.st_mild.ted <- phate.bp (ps_lesional_skin.st_mild.ted, if.invert=FALSE)

#plot cell type distribution along the spacefold axis
#plot.beeswarm.no.pdf(ps_lesional_skin.st_mild.ted, pdf.prefix="PS_LESIONAL_SKIN_MILD_SPACEFOLD_OUTPUT(0.95_cutoff)",q.cut = 0.95,cell.type.order = ps.order)

#plot cell type distribution along the spacefold axis
plot.beeswarm.no.pdf(ps_lesional_skin.st_mild.ted, pdf.prefix="PS_LESIONAL_SKIN_MILD_SPACEFOLD_OUTPUT(0.90_cutoff)",q.cut = 0.90,cell.type.order = ps.order)

```

#### **FIGURE 7A -** NON-LESIONAL & MILD

#### SPACE-FOLD PROJECTION (NON-LESIONAL SKIN - MILD GROUP)

LOAD NON-LESIONAL SKIN DATA - GROUPED BY SEVERITY

```{r}
ps_non_lesional_skin.st_mild.ted <- readRDS("/Volumes/Extreme Pro/GITHUB-DATA/BAYESPRISM-DATA/BAYES_PRISM_OUTPUT/Archive/ps_non_lesional_skin.st_mild.ted.RDS")
ps_non_lesional_skin.st_severe.ted <- readRDS("/Volumes/Extreme Pro/GITHUB-DATA/BAYESPRISM-DATA/BAYES_PRISM_OUTPUT/Archive/ps_non_lesional.st_severe.ted.RDS")
```

```{r}
meta <- as.data.frame(ps_non_lesional_skin.st_mild@meta.data) %>% rownames_to_column("barcode")
rownames(ps_non_lesional_skin.st_mild.ted$para$X) <- meta$barcode
ps_non_lesional_skin.st_mild.ted <- add.meta(bp.obj= ps_non_lesional_skin.st_mild.ted, meta =meta)
```

```{r}
#compute Znk (fraction of reads in each cell type)
ps_non_lesional_skin.st_mild.ted <- compute.Znk(ps_non_lesional_skin.st_mild.ted)

#normalize by Znk
ps_non_lesional_skin.st_mild.ted <- norm.by.sf(ps_non_lesional_skin.st_mild.ted)

#determine background level, using final theta
ps_non_lesional_skin.st_mild.ted <- compute.background.level (ps_non_lesional_skin.st_mild.ted, which.theta='final')

#run phate
# specify anchorCellTypes to select cell types (if you need to held out a group of cell types) 
ps_non_lesional_skin.st_mild.ted <- phate.bp (ps_non_lesional_skin.st_mild.ted, if.invert=FALSE)

#plot cell type distribution along the spacefold axis
#plot.beeswarm.no.pdf(ps_non_lesional_skin.st_mild.ted, pdf.prefix="PS_NON_LESIONAL_SKIN_MILD_SPACEFOLD_OUTPUT(0.95_cutoff)",q.cut = 0.95,cell.type.order = ps.order)

#plot cell type distribution along the spacefold axis
plot.beeswarm.no.pdf(ps_non_lesional_skin.st_mild.ted, pdf.prefix="PS_NON_LESIONAL_SKIN_MILD_SPACEFOLD_OUTPUT(0.90_cutoff)",q.cut = 0.90,cell.type.order = ps.order)
```

## **FIGURE 7B**

### **FIGURE 7B -** LESIONAL & MODERATE-SEVERE

#### SPACE-FOLD PROJECTION (LESIONAL SKIN - MODERATE-SEVERE GROUP)

```{r}
ps_lesional_skin.st_severe.ted$para$feature <- ps_lesional_skin.st_severe.ted$para$X
```

```{r}
meta <- as.data.frame(ps_lesional_skin.st_severe@meta.data) %>% rownames_to_column("barcode")
rownames(ps_lesional_skin.st_severe.ted$para$X) <- meta$barcode
ps_lesional_skin.st_severe.ted <- add.meta(bp.obj= ps_lesional_skin.st_severe.ted, meta =meta)
```

```{r}
#compute Znk (fraction of reads in each cell type)
ps_lesional_skin.st_severe.ted <- compute.Znk(ps_lesional_skin.st_severe.ted)

#normalize by Znk
ps_lesional_skin.st_severe.ted <- norm.by.sf(ps_lesional_skin.st_severe.ted)

#determine background level, using final theta
ps_lesional_skin.st_severe.ted <- compute.background.level (ps_lesional_skin.st_severe.ted, which.theta='final')

#run phate 
# specify anchorCellTypes to select cell types (if you need to held out a group of cell types) 
ps_lesional_skin.st_severe.ted <- phate.bp (ps_lesional_skin.st_severe.ted, if.invert=FALSE)

#plot cell type distribution along the spacefold axis
#plot.beeswarm.no.pdf(ps_lesional_skin.st_severe.ted, pdf.prefix="PS_LESIONAL_SKIN_SEVERE_SPACEFOLD_OUTPUT(0.95_cutoff)",q.cut = 0.95,cell.type.order = ps.order)

#plot cell type distribution along the spacefold axis
plot.beeswarm.no.pdf(ps_lesional_skin.st_severe.ted, pdf.prefix="PS_LESIONAL_SKIN_SEVERE_SPACEFOLD_OUTPUT(0.90_cutoff)",q.cut = 0.90,cell.type.order = ps.order)
```

#### **FIGURE 7B -** NON-LESIONAL & MODERATE-SEVERE

#### SPACE-FOLD PROJECTION

```{r}
meta <- as.data.frame(ps_non_lesional.st_severe@meta.data) %>% rownames_to_column("barcode")
rownames(ps_non_lesional_skin.st_severe.ted$para$X) <- meta$barcode
ps_non_lesional_skin.st_severe.ted <- add.meta(bp.obj= ps_non_lesional_skin.st_severe.ted, meta =meta)
```

```{r}
#compute Znk (fraction of reads in each cell type)
ps_non_lesional_skin.st_severe.ted<- compute.Znk(ps_non_lesional_skin.st_severe.ted)

#normalize by Znk
ps_non_lesional_skin.st_severe.ted<- norm.by.sf(ps_non_lesional_skin.st_severe.ted)

#determine background level, using final theta
ps_non_lesional_skin.st_severe.ted <- compute.background.level (ps_non_lesional_skin.st_severe.ted, which.theta='final')

#run phate
# specify anchorCellTypes to select cell types (if you need to held out a group of cell types) 
ps_non_lesional_skin.st_severe.ted<- phate.bp (ps_non_lesional_skin.st_severe.ted, if.invert=FALSE)

#plot cell type distribution along the spacefold axis
#plot.beeswarm.no.pdf(ps_non_lesional_skin.st_severe.ted, pdf.prefix="PS_NON_LESIONAL_SKIN_SEVERE_SPACEFOLD_OUTPUT(0.95_cutoff)",q.cut = 0.95,cell.type.order = ps.order)     

#plot cell type distribution along the spacefold axis
plot.beeswarm.no.pdf(ps_non_lesional_skin.st_severe.ted, pdf.prefix="PS_NON_LESIONAL_SKIN_SEVERE_SPACEFOLD_OUTPUT(0.90_cutoff)",q.cut = 0.90,cell.type.order = ps.order)
```
