---
title: "Healthy donors - Spatial Transcriptomics data analysis workflow"
output: github_document
---

# ST HEALTHY SAMPLES - PART 1

This notebook covers the basic to advanced analysis of ST (Spatial Transcriptomics) samples from healthy / normal human skin -

-   Pre-processing

-   Quality Control

-   Clustering

-   Dimensionality Reduction (UMAP)

-   Batch Correction

-   Marker Genes

### TABLE OF CONTENTS

| FIGURE NO | LINK         |
|-----------|--------------|
| 1B        | [link](#1b)  |
| 1C        | [link](#1c)  |
| 1D        | [link](#1d)  |
| S3-A      | [link](#s3a) |
| S3-B      | [link](#s3b) |
| S3-C      | [link](#s3c) |
| S2-C      | [link](#s2c) |
| S2-D      | [link](#s2d) |

## Load required packages

```{r}
library(tidyverse)
library(Seurat)
library(cowplot)
library(RColorBrewer)
library(pheatmap)
library(scales)
library(ggsci)
library(harmony)
```

## Import functions from the custom R script

```{r}
source("../SPATIAL_FUNCTIONS.R")
```

### Home directory for processed files

Please replace the home directory location to allow the files to be easily accessible

### Import Healthy Samples

(Raw Data stored in GEO repo: GSE202011)

(Sample meta-data in XXX)

```{r}
## HEALTHY SAMPLES
## HEALTHY MALE SKIN 1 & 2
HEALTHY.Male.s1 <- Load10X_Spatial(data.dir ="/Volumes/Extreme Pro/GITHUB-DATA/ST-DATA/HEALTHY-DATA/Spatial-Data/ST_HM_1/", slice="ST-HM-1-R1")
HEALTHY.Male.s2 <- Load10X_Spatial(data.dir ="/Volumes/Extreme Pro/GITHUB-DATA/ST-DATA/HEALTHY-DATA/Spatial-Data/ST_HM_1_R2/", slice="ST-HM-1-R2")

## HEALTHY FEMALE SKIN 1 & 2 (FROM THE SAME DONOR)
HEALTHY.Female1.s1 <- Load10X_Spatial(data.dir ="/Volumes/Extreme Pro/GITHUB-DATA/ST-DATA/HEALTHY-DATA/Spatial-Data/ST_HF_1/", slice="ST-HF-1-R1")
HEALTHY.Female1.s2 <- Load10X_Spatial(data.dir ="/Volumes/Extreme Pro/GITHUB-DATA/ST-DATA/HEALTHY-DATA/Spatial-Data/ST_HF_1_R2/", slice="ST-HF-1-R2")
HEALTHY.Female1.s3 <- Load10X_Spatial(data.dir ="/Volumes/Extreme Pro/GITHUB-DATA/ST-DATA/HEALTHY-DATA/Spatial-Data/ST_HF_1T/", slice="ST-HF-1-R3")

## HEALTHY FEMALE
HEALTHY.Female2.s1 <- Load10X_Spatial(data.dir ="/Volumes/Extreme Pro/GITHUB-DATA/ST-DATA/HEALTHY-DATA/Spatial-Data/ST_HF_2E/", slice="ST-HF-2-R1")
HEALTHY.Female2.s2 <- Load10X_Spatial(data.dir ="/Volumes/Extreme Pro/GITHUB-DATA/ST-DATA/HEALTHY-DATA/Spatial-Data/ST_HF_2T/", slice="ST-HF-2-R2")
```

### Import the spots to be removed.

(Outlier spots identified using visual inspection in loupe browser)

Outlier spots here are spots that visually do not appear to be part of the tissue slice

```{r}
# HEALTHY FEMALE SKIN 2
# REMOVE SPOTS
remove.spots <- read.csv(file="/Volumes/Extreme Pro/GITHUB-DATA/ST-DATA/HEALTHY-DATA/Remove-Spots-Files/HEALTHY.Female1.s2.remove.spots.csv")
subset_spots <- Cells(HEALTHY.Female1.s2)[which((!(rownames(HEALTHY.Female1.s2@meta.data) %in% remove.spots$Barcode)))]
HEALTHY.Female1.s2.clean <- subset(HEALTHY.Female1.s2,cells=subset_spots)
```

### SAMPLE IDS

(HV - Healthy Volunteer, R - Replicate)

```{r}
HEALTHY.Male.s1$sample.id<- "HV1_S1_R1" 
HEALTHY.Male.s2$sample.id<- "HV1_S1_R2"

HEALTHY.Female1.s1$sample.id <- "HV2_S1_R1"
HEALTHY.Female1.s2.clean$sample.id <- "HV2_S1_R2"
HEALTHY.Female1.s3$sample.id <- "HV2_S2"

HEALTHY.Female2.s1$sample.id <- "HV3_S1"
HEALTHY.Female2.s2$sample.id <- "HV3_S2"
```

### SAMPLE IDS (version 2) - (Full name + Biopsy site)

Biopsy sites - Trunk and FA (Forearm)

```{r}
HEALTHY.Male.s1$orig.ident <- "Healthy Volunteer 1 - TRUNK R1" 
HEALTHY.Male.s2$orig.ident <-"Healthy Volunteer 1 - TRUNK R2"

HEALTHY.Female1.s1$orig.ident <- "Healthy Volunteer 2 - FA R1"
HEALTHY.Female1.s2.clean$orig.ident <- "Healthy Volunteer 2 - FA R2"
HEALTHY.Female1.s3$orig.ident <- "Healthy Volunteer 2 - TRUNK"

HEALTHY.Female2.s1$orig.ident <- "Healthy Volunteer 3 - FA R1"
HEALTHY.Female2.s2$orig.ident <- "Healthy Volunteer 3 - TRUNK"
```

### Combine all samples into one list

```{r}
Healthy_Samples <- c(HEALTHY.Male.s1, HEALTHY.Male.s2, HEALTHY.Female1.s1, HEALTHY.Female1.s2.clean, HEALTHY.Female1.s3, HEALTHY.Female2.s1, HEALTHY.Female2.s2)
```

### Standard spatial plots (before filtering)

Generated per sample

```{r}
for (x in Healthy_Samples){
  st_plot(x)
}
```

### QC (Quality control) plots for each healthy sample

```{r}
for (x in Healthy_Samples){
  st_plot_QC(x)
}
```

### Standard QC scatter plots

(**UMI** / **nCount Spatial** vs. **Number of genes** / **features expressed**)

Pre-set limits for X and Y axis - (12500,8500)

```{r}
for (x in Healthy_Samples){
  st_scatter_QC(x)
}
```

### QC filtering

Remove spots with low depth (Defined by number of genes expressed per spot)

\- 200 genes minimum per spot

```{r}
i <- 1
while(i <= length(Healthy_Samples)){
  filtered_data <- st_filter_by_genes(st.data = Healthy_Samples[[i]],x = 200)
  Healthy_Samples[[i]] <- filtered_data
  i <- i+1
}
```

### Batch correction - Using Seurat Anchor integration

Original vignette (<https://satijalab.org/seurat/articles/integration_introduction.html>)

Reference paper- (<https://doi.org/10.1016/j.cell.2019.05.031>)

Each sample - replicate is treated as an individual batch to account for differences in sequencing depth between replicates and sample to sample heterogeneity in human patient samples.

```{r,eval=FALSE}
new.skin.combined <- st_combine(Healthy_Samples, ndim = 20, res = 0.6)
```

```{r,eval=FALSE}
# Save new.skin.combined object to RDS file
saveRDS(new.skin.combined, "/Volumes/Extreme Pro/GITHUB-DATA/ST-DATA/HEALTHY-DATA/RDS-Files/HEALTHY_SKIN_SAMPLES_ST.RDS")
```

### Load the pre-processed data

(The RDS was processed using the same parameters that have been defined above)

```{r}
new.skin.combined <- readRDS("/Volumes/Extreme Pro/GITHUB-DATA/ST-DATA/HEALTHY-DATA/RDS-Files/HEALTHY_SKIN_SAMPLES_ST.RDS")
```

### Defining and viewing color scheme for clustering based UMAP and spatial plots

```{r}
# Define Color Scheme
custom_colors <- c("#B4DFFC","#6EAB3D","#FFD700","#A020F0","#FFA500","#AEDD3C","#595959","#D2AF81FF","#3A49FC","#FF0000","#A86F3D","#A18FAA")

# View Color Scheme
show_col(custom_colors)
```

### Finding marker genes for batch corrected healthy samples + plotting for top 8 marker genes per cluster heat-map

```{r}
# Find top 8 marker genes for each cluster for heat map
new.skin.combined <- PrepSCTFindMarkers(new.skin.combined,assay = "SCT")
Idents(new.skin.combined) <- "seurat_clusters"
new.skin.combined.markers <- FindAllMarkers(new.skin.combined, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25, assay = "SCT")

DefaultAssay(new.skin.combined) <- "SCT"
top8 <- new.skin.combined.markers %>%
    dplyr::group_by(cluster) %>%
    dplyr::filter(p_val_adj < 0.05) %>%
    top_n(n = 8, wt = avg_log2FC)
```

```{r,eval=FALSE}
# Save Healthy Skin Markers RDS File (Used later in Healthy Samples Part 2 & 4)
saveRDS(new.skin.combined.markers, "/Volumes/Extreme Pro/GITHUB-DATA/ST-DATA/HEALTHY-DATA/RDS-Files/HEALTHY_ST_MARKERS.RDS")
```

<a id="s2c"></a>

```{r figure-s2-c,fig.height=15,fig.width=12}
DoHeatmap(new.skin.combined, features = top8$gene,assay = "SCT", group.colors = custom_colors) + NoLegend()
```

</a>

### Assign spatial region specific cluster ids

```{r,eval=TRUE}
new.cluster.ids <- c("0 Mixed","1 Dermis","2 Dermis Adipose","3 Epidermis 1","4 Dermis Connective Tissue",
                     "5 Upper follicle (UF) and Perifollicular dermis","6 Epidermis 2","7 Dermis 2",
                     "8 Dermis Smooth Muscle","9 Hair follicle","10 Dermis Lymphatics","11 Dermis vasculature")

names(new.cluster.ids) <- levels(new.skin.combined)
```

```{r}
new.skin.combined <- RenameIdents(new.skin.combined, new.cluster.ids)
new.skin.combined <- StashIdent(new.skin.combined, save.name = "Spatial.regions")
```

### Save RDS file skin.new.combined with cluster ids

```{r}
#saveRDS(new.skin.combined, "/Volumes/Extreme Pro/GITHUB-DATA/ST-DATA/HEALTHY-DATA/RDS-Files/HEALTHY_SKIN_SAMPLES_ST_CLUSTERS_LABELED.RDS")
```

```{r,eval=FALSE}
new.skin.combined <- readRDS("/Volumes/Extreme Pro/GITHUB-DATA/ST-DATA/HEALTHY-DATA/RDS-Files/HEALTHY_SKIN_SAMPLES_ST_CLUSTERS_LABELED.RDS")
```

```{r}
# DEFINE COLORS FOR EACH SPATIAL REGION
color.labels.anchor <- c("#B4DFFC","#6EAB3D","#FFD700","#A020F0","#FFA500","#AEDD3C","#595959","#D2AF81FF","#3A49FC","#FF0000","#A86F3D","#A18FAA")
names(color.labels.anchor) <- new.cluster.ids
```

### General UMAP with defined color scheme

#### FIGURE 1-C

<a id="1c">

```{r figure-1c}
#pdf(width = 12, height=8, file = "UMAP_HEALTHY_SAMPLES_CLUSTERS_ONLY.pdf")
DimPlot(new.skin.combined, cols = custom_colors, pt.size = 3.5)
#dev.off()

# FOR IN NOTEBOOK VIEW
DimPlot(new.skin.combined, cols = custom_colors, pt.size = 3.5)
```

</a>

### UMAP plot split by samples

```{r}
DimPlot(new.skin.combined, split.by = "sample.id")
```

### Spatial plot for HV2_S1_R2 / Healthy Volunteer 2 -

```{r}
#pdf(width = 8,height=10,file = "HEALTHY_FEMALE_SAMPLE_1_R2_SPATIAL_PLOT.pdf")
SpatialDimPlot(new.skin.combined,images="HV2.S1.R2",cols = color.labels.anchor,pt.size.factor = 2.4)
#dev.off()
```

### VIEW SPATIAL PLOT HERE

**FIGURE 1-B**

```{r}
names(new.skin.combined@images) <- c("HV1.S1.R1","HV1.S1.R2","HV2.S1.R1","HV2.S1.R2","HV2.S1.R2","HV3.S1","HV3.S2")
```

<a id="1b">

```{r figure-1b}
SpatialDimPlot(new.skin.combined, images="HV2.S1.R2", cols = color.labels.anchor, pt.size.factor = 2.4)
```

</a>

### Generate Spatial plot for all samples

```{r,eval=FALSE}
images <- Images(new.skin.combined) %>% as.vector()
for(x in images){
  pdf(width = 8, height = 10, file = paste(x, "_SPATIAL_PLOT.pdf"))
  print(SpatialDimPlot(new.skin.combined, images=x,cols = custom_colors, pt.size.factor = 2))
  dev.off()
}
```

### Pathway analysis - Performed per cluster

```{r,eval=FALSE}
for(i in seq(0,11)){
  cluster <- new.skin.combined.markers %>% filter(cluster==i)
  GO_PATHWAYS_ENRICH(cluster, ONT="BP", OUTPUT = i)
  GO_PATHWAYS_ENRICH(cluster, ONT="CC", OUTPUT = i)
  GO_PATHWAYS_ENRICH(cluster, ONT="MF", OUTPUT = i)
  KEGG_PATHWAYS(cluster, OUTPUT = i)
}
```

### Violin Plot: UMI counts per cluster basis

<a id="s2d">

```{r figure-s2-d,eval=TRUE}
#pdf(file = "VIOLIN_PLOT_UMIs_and_CLUSTERS_HEALTHY_SAMPLES.pdf",width = 16,height=12)
VlnPlot(new.skin.combined, group.by = "seurat_clusters", features = "nCount_Spatial", cols = custom_colors)
#dev.off()
```

</a>

# LOG TRANSFORMED UMI COUNTS

```{r}
#pdf(file = "VIOLIN_PLOT_UMIs_and_CLUSTERS_HEALTHY_SAMPLES(LOG_TRANSFORMED).pdf",width = 16,height=12)
VlnPlot(new.skin.combined,group.by = "seurat_clusters", features = "nCount_Spatial", cols = custom_colors, log = TRUE)
#dev.off()
```

```{r}
# SPOT FREQUENCY TABLE - BY SPATIAL REGION
spatial.freq <- as.data.frame(table(new.skin.combined@meta.data$Spatial.regions))

# Write Spatial Frequency Results to CSV
#write.csv(spatial.freq, file="SPATIAL_REGIONS_CLUSTERS_SPOT_COUNTS.csv")
```

## AVERAGE GENE COUNTS

```{r}
Idents(new.skin.combined) <- "seurat_clusters"

clusters <- unique(new.skin.combined@meta.data$seurat_clusters) %>% as.vector()

for(x in clusters){
  avg_exp.cluster <- AverageExpression(new.skin.combined,slot = "counts", assay="Spatial") %>% 
    as.data.frame() %>%
    dplyr::select(paste("Spatial.",x,sep = "")) %>% 
    filter(x>0) %>% 
    write.csv(file = paste("CLUSTER_",x,"_GENE_COUNTS.csv",sep = ""))
}
```

## **FIGURE 1D**

## PERCENTAGE PLOT FOR DIFFERENT SPATIAL REGIONS

<a id="1d">

```{r,fig.height=10,fig.width=10}
Regions.df <- table(new.skin.combined@meta.data$Spatial.regions, new.skin.combined@meta.data$sample.id) %>% as.data.frame() %>% dplyr::rename(Spatial_Region=Var1, Sample=Var2) 

black.bold.16.text <- element_text(face = "bold", color = "black", size = 14,angle = 90, vjust = 0.5, hjust=1)

brks <- c(0, 0.25, 0.5, 0.75, 1)

#pdf(file="PERCENTAGE_COMPOSTION_PLOT_HEALTHY_SAMPLES.pdf",height = 14,width = 8)
ggplot(Regions.df,aes(x=Sample,y=Freq,fill=Spatial_Region)) + geom_bar(stat="identity", position="fill") + scale_fill_manual(values =color.labels.anchor)  + ggplot2::theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.text.x=black.bold.16.text) + scale_y_continuous(breaks = brks, labels = scales::percent(brks)) + ylab("% Composition of Sample by Spatial Region / Clusters")
#dev.off()
```

</a>

# Harmony Batch Correction

### Merge Samples, Run PCA & Harmony Batch Correction

```{r}
# Healthy_Samples was generated as list of Samples at the top of the script
healthy_skin.merge <- merge(x=Healthy_Samples[[1]], y=Healthy_Samples[-1])
healthy_skin.merge <- SCTransform(healthy_skin.merge, assay = "Spatial", vars.to.regress = c("orig.ident"))

DefaultAssay(healthy_skin.merge) <- "SCT"
healthy_skin.merge <- RunPCA(healthy_skin.merge)

all_healthy_samples.hm.sct <- RunHarmony(healthy_skin.merge, assay.use = "SCT", project.dim = FALSE, group.by.vars = "sample.id")
```

### View Elbow Plot to Determine Dims for UMAP

```{r}
ElbowPlot(all_healthy_samples.hm.sct)
```

### Run UMAP & Clustering

```{r}
all_healthy_samples.hm.sct <- all_healthy_samples.hm.sct %>% 
    RunUMAP(reduction = "harmony", dims = 1:20) %>% 
    FindNeighbors(reduction = "harmony", dims = 1:20) %>% 
    FindClusters(resolution = 0.35) %>% 
    identity()

# Save Harmonized Healthy Samples
#saveRDS(all_healthy_samples.hm.sct, "/Volumes/Extreme Pro/GITHUB-DATA/ST-DATA/HEALTHY-DATA/RDS-Files/HEALTHY_ST_(Harmonized_for_final_figure).RDS")
```

### Load Harmonized Healthy Samples

```{r}
all_healthy_samples.hm.sct <- readRDS("/Volumes/Extreme Pro/GITHUB-DATA/ST-DATA/HEALTHY-DATA/RDS-Files/HEALTHY_ST_(Harmonized_for_final_figure).RDS")
```



<a id="s3a">

### Find All Markers for Harmonized Healthy Samples

```{r ,fig.height=10, fig.width=12}
# Find All Healthy Harmonized Cluster Markers
Idents(all_healthy_samples.hm.sct) <- "seurat_clusters"
all_healthy_samples.hm.sct.markers <- FindAllMarkers(all_healthy_samples.hm.sct, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25,assay = "SCT")

# Get Top 8 genes from each Cluster
DefaultAssay(all_healthy_samples.hm.sct) <- "SCT"
top8 <- all_healthy_samples.hm.sct.markers %>%
    dplyr::group_by(cluster) %>%
    dplyr::filter(p_val_adj<0.05) %>%
    top_n(n = 8, wt = avg_log2FC)

#pdf(width = 20,height=15,file = "HEATMAP_HEALTHY_SAMPLES_(HM_VER_FINAL).pdf")
DoHeatmap(all_healthy_samples.hm.sct, features = top8$gene,assay = "SCT", group.colors = custom_colors) + NoLegend()
#dev.off()
```

</a>

<a id="s3b">
### View UMAP

```{r}
#pdf(width = 12,height=8,file = "UMAP_HEALTHY_SAMPLES_(HM_VER_FINAL).pdf")
DimPlot(all_healthy_samples.hm.sct, pt.size = 3.5, cols = custom_colors)
#dev.off()
```

### Percentage composition plot

```{r,fig.height=10,fig.width=10}
Regions.df <- table(all_healthy_samples.hm.sct@meta.data$seurat_clusters, all_healthy_samples.hm.sct@meta.data$sample.id) %>% 
  as.data.frame() %>% 
  dplyr::rename(Spatial_Region=Var1,Sample=Var2) 

black.bold.16.text <- element_text(face = "bold", color = "black", size = 14,angle = 90, vjust = 0.5, hjust=1)

brks <- c(0, 0.25, 0.5, 0.75, 1)

#pdf(file="PERCENTAGE_COMPOSTION_PLOT_HEALTHY_SAMPLES(AFTER_HM_CORRECTION).pdf",height = 14,width = 8)
ggplot(Regions.df,aes(x=Sample,y=Freq,fill=Spatial_Region)) + geom_bar(stat="identity", position="fill") + ggplot2::theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.text.x = black.bold.16.text) + 
  scale_y_continuous(breaks = brks, labels = scales::percent(brks)) + 
  ylab("% Composition of Sample by Spatial Region / Clusters") +  
  scale_fill_manual(values = custom_colors)
#dev.off()
```

```{r}
#pdf(width = 8,height=10,file = "HEALTHY_FEMALE_SAMPLE_1_R2_SPATIAL_PLOT(HM_VERSION_FINAL).pdf")
SpatialDimPlot(all_healthy_samples.hm.sct, images="HV2.S1.R2", pt.size.factor = 2.4) + 
  scale_fill_manual(values = custom_colors) 
#dev.off()
```

</a>

## Alternative approach

```{r}
for (i in 1:length(Healthy_Samples)) {
    Healthy_Samples[[i]] <- SCTransform(Healthy_Samples[[i]], verbose = FALSE, assay="Spatial")
}

all.variable.features <- c()

for(x in Healthy_Samples){
  x <- FindVariableFeatures(x)
  vf <- VariableFeatures(x)
  all.variable.features <- c(all.variable.features, vf) 
}
```

```{r,warning=FALSE}
# Merge Samples
healthy_skin.merge <- merge(x=Healthy_Samples[[1]], y=Healthy_Samples[-1])

# Find Variable Features & Run Harmony
VariableFeatures(healthy_skin.merge) <- all.variable.features
DefaultAssay(healthy_skin.merge) <- "SCT"
healthy_skin.merge <- PrepSCTFindMarkers(healthy_skin.merge)
healthy_skin.merge <- RunPCA(healthy_skin.merge)
all_healthy_samples.hm.sct.alt <- RunHarmony(healthy_skin.merge, assay.use = "SCT", project.dim = FALSE,group.by.vars = "sample.id")
```

```{r}
all_healthy_samples.hm.sct.alt <- all_healthy_samples.hm.sct.alt %>% 
    RunUMAP(reduction = "harmony", dims = 1:40) %>% 
    FindNeighbors(reduction = "harmony", dims = 1:40) %>% 
    FindClusters(resolution = 0.6) %>% 
    identity()
```

```{r,eval=FALSE}
#saveRDS(all_healthy_samples.hm.sct.alt, file="/Volumes/Extreme Pro/GITHUB-DATA/ST-DATA/HEALTHY-DATA/RDS-Files/HEALTHY_ST(Harmonized_Alt_Approach).RDS")
```

## Alternative Harmony approach (Not used in the final manuscript)

### Load RDS File

```{r}
all_healthy_samples.hm.sct.alt <- readRDS("/Volumes/Extreme Pro/GITHUB-DATA/ST-DATA/HEALTHY-DATA/RDS-Files/HEALTHY_ST(Harmonized_Alt_Approach).RDS")
```

### Colors of Alternative Downstream Plots

```{r}
custom_colors_alt <- c("#B4DFFC","#6EAB3D","#FFD700","#A020F0","#FFA500","#AEDD3C","#595959","#D2AF81FF","#3A49FC")
show_col(custom_colors_alt)
```

### Spatial DimPlots All Samples

```{r,eval=FALSE}
#dir.create("HM_CORRECTED_HEALTHY_SAMPLES/")

images <- Images(all_healthy_samples.hm.sct.alt) %>% as.vector()
for(x in images){
  #pdf(width = 8,height=10,file = paste("HM_CORRECTED_HEALTHY_SAMPLES/",x,"_(HM_CORRECTED)_HEALTHY_SPATIAL_PLOT.pdf"))
  print(SpatialDimPlot(all_healthy_samples.hm.sct.alt, images=x, pt.size.factor = 2) +
          scale_fill_manual(values = custom_colors_alt))
  #dev.off()
}
```

### Top Marker Genes Heat Map

```{r ,fig.height=12, fig.width=10}
all_healthy_samples.hm.sct.alt <- PrepSCTFindMarkers(all_healthy_samples.hm.sct.alt,assay = "SCT")
Idents(all_healthy_samples.hm.sct.alt) <- "seurat_clusters"

all_healthy_samples.hm.sct.alt.markers <- FindAllMarkers(all_healthy_samples.hm.sct.alt, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25,assay = "SCT")

DefaultAssay(all_healthy_samples.hm.sct.alt) <- "SCT"
top8 <- all_healthy_samples.hm.sct.alt.markers %>%
    dplyr::group_by(cluster) %>%
    dplyr::filter(p_val_adj<0.05) %>%
    top_n(n = 8, wt = avg_log2FC)

#pdf(width = 20,height=15,file = "HEATMAP_HEALTHY_SAMPLES_(HM_VER_1).pdf")
print(DoHeatmap(all_healthy_samples.hm.sct.alt, features = top8$gene,assay = "SCT", group.colors = custom_colors_alt) + NoLegend())
#dev.off()
```

### Assign Cluster Labels

```{r}
cluster.labels <- c("0 Dermis fibroblasts", "1 Mixed", "2 Dermis adipose, connective tissue", "3 Macrophages", "4 Epidermis 1", "5 Smooth muscle cells", "6 Epidermis 2", "7 Melanocytes", "8 Dermal lymphatics")
names(cluster.labels) <- levels(all_healthy_samples.hm.sct.alt)

all_healthy_samples.hm.sct.alt <- RenameIdents(all_healthy_samples.hm.sct.alt, cluster.labels)
all_healthy_samples.hm.sct.alt[["Spatial.regions"]] <- Idents(object = all_healthy_samples.hm.sct.alt)
```

### Plot UMAP

```{r}
#pdf(width = 12,height=8,file = "UMAP_HEALTHY_SAMPLES_(HM_VER_1).pdf")
DimPlot(all_healthy_samples.hm.sct.alt,cols = custom_colors_alt, pt.size = 3.5)
#dev.off(){r}
```

### Percentage plot for different spatial regions

```{r,fig.height=10,fig.width=10}
Regions.df <- table(all_healthy_samples.hm.sct.alt@meta.data$Spatial.regions,all_healthy_samples.hm.sct.alt@meta.data$sample.id) %>% 
  as.data.frame() %>% 
  dplyr::rename(Spatial_Region=Var1,Sample=Var2) 

black.bold.16.text <- element_text(face = "bold", color = "black", size = 14,angle = 90, vjust = 0.5, hjust=1)

brks <- c(0, 0.25, 0.5, 0.75, 1)

#pdf(file="HM_CORRECTED_HEALTHY_SAMPLES/PERCENTAGE_COMPOSTION_PLOT_HEALTHY_SAMPLES(AFTER_HM_CORRECTION).pdf",height = 14,width = 8)
ggplot(Regions.df,aes(x=Sample,y=Freq,fill=Spatial_Region)) + geom_bar(stat="identity",position="fill") + scale_fill_manual(values = custom_colors_alt)  + 
  ggplot2::theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
                 panel.background = element_blank(),axis.text.x = black.bold.16.text) + 
  scale_y_continuous(breaks = brks, labels = scales::percent(brks)) + 
  ylab("% Composition of Sample by Spatial Region / Clusters")
#dev.off()
```
